dev-system/dev/user.clj

```diff
@@ -35,6 +35,66 @@
           (fn [app-map]
             (u/map-values #(assoc-in % [:log :level] level) app-map))))
 
+(defn- maybe-set-all-modes
+  "A utility function for (conditionally) setting all the values of a map to
+  the same value."
+  [all new-modes]
```

Comment:
Hrm, yeah. Definitely a bit vague. Also good point about the side effects ... the function probably doesn't merit a `!`, since sometimes it will not change anything. So, yeah, I should make up for this by mentioning the side effect nature ...